typescript/sdk/src/middleware/tokenbridge.hardhat-test.ts

```diff
@@ -0,0 +1,109 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+
+import {
+  MockToken,
+  MockToken__factory,
+  TestTokenBridgeMessageRecipient__factory,
+  TokenBridgeRouter,
+} from '@hyperlane-xyz/core';
+import { utils } from '@hyperlane-xyz/utils';
+
+import { testChainConnectionConfigs } from '../consts/chainConnectionConfigs';
+import { TestCoreApp } from '../core/TestCoreApp';
+import { TestCoreDeployer } from '../core/TestCoreDeployer';
+import { TokenBridgeApp } from '../deploy/middleware/TokenBridgeApp';
+import {
+  BridgeAdapterType,
+  TokenBridgeConfig,
+  TokenBridgeDeployer,
+} from '../deploy/middleware/TokenBridgeRouterDeployer';
+import { getChainToOwnerMap, getTestMultiProvider } from '../deploy/utils';
+import { ChainNameToDomainId } from '../domains';
+import { MultiProvider } from '../providers/MultiProvider';
+import { ChainMap, TestChainNames } from '../types';
+import { objMap } from '../utils/objects';
+
+describe('TokenBridgeRouter', async () => {
+  const localChain = 'test1';
+  const remoteChain = 'test2';
+  const remoteDomain = ChainNameToDomainId[remoteChain];
+
+  let signer: SignerWithAddress;
+  let local: TokenBridgeRouter;
+  let multiProvider: MultiProvider<TestChainNames>;
+  let coreApp: TestCoreApp;
+
+  let tokenBridgeApp: TokenBridgeApp<TestChainNames>;
+  let config: ChainMap<TestChainNames, TokenBridgeConfig>;
+  let mockToken: MockToken;
+
+  before(async () => {
+    [signer] = await ethers.getSigners();
+
+    multiProvider = getTestMultiProvider(signer);
+
+    const coreDeployer = new TestCoreDeployer(multiProvider);
+    const coreContractsMaps = await coreDeployer.deploy();
+    coreApp = new TestCoreApp(coreContractsMaps, multiProvider);
+
+    const mockTokenF = new MockToken__factory(signer);
+    mockToken = await mockTokenF.deploy();
+
+    config = coreApp.extendWithConnectionClientConfig(
```

Comment:
these are getting a bit illegible

Reply:
idk if there is an easier way though?

Reply:
If/when we overhaul the SDK, one idea would be a set of functions with consistent naming that extend configs with different things. e.g. `withClientConfig`, `withOwnerAddress`, etc.